### PR TITLE
Lep 135 self employment design

### PR DESCRIPTION
**Pull request by starswan**
https://dsdmoj.atlassian.net/browse/LEP-135

Design for self employment inputs and outputs in swagger docs

### CODE REVIEW

File: spec/swagger_helper.rb [hunk 1]
```diff
@@ -698,6 +698,74 @@
               amount: { "$ref" => "#/components/schemas/currency" },
             },
           },
+          SelfEmploymentType: {
+            type: :string,
+            enum: %i[partnership company_director sole_trader],
```
Comment (davidread) [inline spec/swagger_helper.rb:703]: This misses off LLP. There may be other business structures too.

I wonder if we can simplify this by just accepting two clearer answers?:
* 'the client is responsible for the debts of the business' - ie sole trader OR partner in a (normal) partnership
* 'the business is a separate entity from the client' - ie company director of limited company OR partner of a LLP

Comment (starswan) [reply spec/swagger_helper.rb:703]: @davidread not according to the Miro - all different types (apparently) have different calculations

File: spec/swagger_helper.rb [hunk 2]
```diff
@@ -698,6 +698,74 @@
...
+          },
+          SelfEmployment: {
+            type: :object,
+            required: %i[name self_employment_type net_profit_or_drawings percentage_owned business_capital],
+            additionalProperties: false,
+            properties: {
+              name: { type: :string },
+              self_employment_type: { "$ref" => "#/components/schemas/SelfEmploymentType" },
+              net_profit_or_drawings: {
```
Comment (davidread) [inline spec/swagger_helper.rb:712]: what about income tax and NI?

File: spec/swagger_helper.rb (hunk 1 above)
Comment (davidread) [reply spec/swagger_helper.rb:703]: ok makes sense, let's keep them like this then

File: spec/swagger_helper.rb (hunk 2 above)
Comment (starswan) [reply spec/swagger_helper.rb:712]: @davidread I'm assuming that income tax and national insurance i will be submitted with 'employment income' (along with benefits_in_kind). The employers NI doesn't look like it comes into the equation, its all subsumed into this 'net_profit' figure

**Pull request closed without merging**